File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        
        int l = 0;
        int r = nums.size() - 1;
        int ans = 1000;
        while(l<=r){
            cout<<nums[l]<<" "<<nums[r]<<"\n";
            if(nums[l] > nums[r]){
                ans = min(ans,nums[r]);
                r--;
                l++;
            }
            else{
                ans = min(nums[l],ans);
                break;
            }
            cout<<ans<<"\n";
        }
        return ans;
    }
};

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        q = deque()
        q.append(root)
        ans = []
        while q:
            qlen = len(q)
            for i in range(qlen):
                node = q.popleft()
                if i == qlen - 1:
                    ans.append(node.val)
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)            

        return ans
        